types/address.go

```diff
@@ -103,6 +86,19 @@ func (a *Address) Decode(decoder scale.Decoder) error {
 		return err
 	}
 
+	if decoder.GetOpts().SkipAccountIDHeader {
```

Comment:
@mikiquantum this check should be the first – otherwise you might end up with an `AccountIndex` but you wanted to "skip", which implicitly asserts that there are no indices. 

I still feel that we should rename `SkipAccountIDHeader` to what is actually implied, i. e. `UseAccountIndex` or `HasPalletIndices`.

Reply:
Yes you are right, will change.